### PR TITLE
Zelin/#2 Added Implementation of PortalChooserPage

### PR DESCRIPTION
**Pull request by VBZelin**
## Issue

Resolves issue #2 

## Root cause

- The app needs a way to allow the user to sign in with various portal urls 

## Solution description

- Added support for checking / filtering / saving / reading ArcGIS Online portal
- Added `ArcGISRuntimeService` as dependency injection
- Added `SQLDatabaseService` as dependency injection
- Added Esri .Net Map SDK NuGet package
- Added sqlite-net-pcl NuGet package
- Added Json.Net Newtonsoft NuGet package

## Notice
- OAuth sign in is not finished yet, will be finished in #1 
- Current implementation only supports checking ArcGIS Online portals, will need different issues for supporting IWA and PKI portals

## Screenshots
Screenshot_1674961745

## Checklist

- [x] This pull request has been linked to the issue being resolved

### Compiled for:

- [x] Android
- [x] iOS

### Tested on:

- [ ] Windows
- Android
    - [x] Emulator
    - [ ] Device
- iOS
    - [x] Simulator
    - [ ] Device

### CODE REVIEW

File: ArcGIS.StoryMaps.BriefingBook/Pages/PortalChooserPage/PortalChooserPage.xaml [hunk 1]
```diff
@@ -3,86 +3,76 @@
...
-
-        <Entry
-             x:Name="entry"
+		<Grid.RowDefinitions>
+			<RowDefinition Height="Auto" />
+			<RowDefinition Height="Auto" />
+			<RowDefinition Height="*" />
+		</Grid.RowDefinitions>
+		<Entry
+             x:Name="Entry"
```
Comment (Tframe) [inline ArcGIS.StoryMaps.BriefingBook/Pages/PortalChooserPage/PortalChooserPage.xaml:23]: Code indentation starting at Entry x:Name looks off.

File: ArcGIS.StoryMaps.BriefingBook/Helpers/Utility.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using Newtonsoft.Json;
+
+namespace ArcGIS.StoryMaps.BriefingBook.Helpers
+{
+    public static class Utility
+    {
+        // Convert url to portal url starts with 'https://'
```
Comment (Tframe) [inline ArcGIS.StoryMaps.BriefingBook/Helpers/Utility.cs:9]: Do we want to use this comment format for methods? 
`        /// <summary>
        /// Parameters
        /// </summary>`

**Review by Tframe — COMMENTED** [review]
Code looks great. Only a couple things I noticed. Method comments had inconsistent formats and one xaml page appeared to have indentation issues.

File: ArcGIS.StoryMaps.BriefingBook/Helpers/Utility.cs (hunk 2 above)
Comment (VBZelin) [reply ArcGIS.StoryMaps.BriefingBook/Helpers/Utility.cs]: I think this format is for describing method, inside we can still use`//`.

Comment (VBZelin) [reply ArcGIS.StoryMaps.BriefingBook/Helpers/Utility.cs]: This has been resolved in the latest commit.

**Pull request merged**